gulpfile.js

```diff
@@ -0,0 +1,386 @@
+    replace = require('gulp-replace');
+
+var config = require('./package.json');
+var settings = config.settings;
+    settings.liveReload=false;
+    settings.plumberConfig=function(){
+      return {'errorHandler': onError};
+    };
+
+/**
+ * browser-sync task for starting a server. This will open a browser for you. Point multiple browsers / devices to the same url and watch the magic happen.
+ * Depends on: watch
+ */
+gulp.task('browser-sync', ['watch'], function() {
+  var browserSync = require('browser-sync');
+
+  // Watch any files in dist/*, reload on change
+  gulp.watch([settings.dist + '**']).on('change', function(){browserSync.reload({});notify({ message: 'Reload browser' });});
+
+  return browserSync({
+      server: {
+          baseDir: settings.dist
+      },
+      ghostMode: {
+        clicks: true,
+        location: true,
+        forms: true,
+        scroll: true
+      },
+      open: "external",
+      injectChanges: true, // inject CSS changes (false force a reload) 
+      browser: ["google chrome"],
+      scrollProportionally: true, // Sync viewports to TOP position
+      scrollThrottle: 50,
+    });
+});
+
+
+/**
+ * Build and copy all styles, scripts, images and fonts.
+ * Depends on: clean
+ */
+gulp.task('build', ['info', 'clean'], function() {
+  gulp.start('styles', 'scripts', 'images', 'copy', 'todo');
+});
+
+
+/**
+ * Cleans the `dist` folder and other generated files
+ */
+gulp.task('clean', ['clear-cache'],  function(cb) {
+  del([settings.dist, 'todo.md', 'todo.json'], cb);
+});
+
+/**
+ * Clears the cache used by gulp-cache
+ */
+gulp.task('clear-cache', function() {
+  // Or, just call this for everything
+  cache.clearAll();
+});
+
+
+/**
+ * Copies all to dist/
+ */
+gulp.task('copy', ['copy-fonts', 'copy-template', 'copy-index'], function() {});
+
+
+/**
+ * Task for copying fonts only
+ */
+gulp.task('copy-fonts', function() {
+  var deferred = q.defer();
+   // copy all fonts
+   setTimeout(function() {
+    gulp.src( settings.src + 'fonts/**')
+      .pipe(gulp.dest(settings.dist + 'fonts'));
+       deferred.resolve();
+  }, 1);
+
+  return deferred.promise;
+});
+
+
+/**
+ * task for copying templates only
+ */
+gulp.task('copy-template', function() {
+  // copy all html && json
+  return gulp.src( [settings.src + 'js/app/**/*.html', settings.src + 'js/app/**/*.json'])
+    .pipe(cache(gulp.dest('dist/js/app')));
+});
+
+/**
+ * Task for copying index page only. Optionally add live reload script to it
+ */
+gulp.task('copy-index', function() {
+   // copy the index.html
+   return gulp.src(settings.src + 'index.html')
+    .pipe(gulpif(argv.dev, replace(/app.min.js/g, 'app.js')))
+    .pipe(gulpif(argv.nohuna, replace('<script src=\'js/huna.min.js\'></script>', '')))
+    .pipe(gulpif(settings.liveReload, replace(/(\<\/body\>)/g, "<script>document.write('<script src=\"http://' + (location.host || 'localhost').split(':')[0] + ':35729/livereload.js?snipver=1\"></' + 'script>')</script>$1")))
```

Comment:
Line is too long.<br>Unexpected escaped character '<' in regular expression.<br>Mixed double and single quotes.